### ISSUE TITLE
ERROR:dbus.proxies:Introspect error on :1.17:/org/freedesktop/sssd/infopipe/Users: dbus.exceptions.DBusException: org.freedesktop.DBus.Error.AuthFailed: Permission denied

### ISSUE THREAD

**Issue by adelton** [adelton/webauthinfra#6]
On https://github.com/adelton/webauthinfra/tree/fedora-29 branch, www fails with

```
www_1     | + ipsilon-client-install --saml-idp-metadata https://idp.example.test/idp/saml2/metadata --saml-auth /app/saml-login --saml-sp-name app --saml-idp-url https://idp.example.test/idp/ --saml-sp-name app --admin-password /tmp/admin-password --no-saml-soap-logout
www_1     | --saml-auth is deprecated. Please use --auth-location
www_1     | Generating a RSA private key
www_1     | .................................................+++++
www_1     | .+++++
www_1     | writing new private key to '/etc/httpd/saml2/www.example.test/certificate.key'
www_1     | -----
www_1     | Failed to register SP with IDP!
www_1     | Error: [Unable to authenticate to IdP (500)]
www_1     | Unable to authenticate to IdP (500)
```

On idp in /var/log/httpd/ssl_error_log, there is

```
[Wed Dec 12 06:19:46.274288 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39944] [12/Dec/2018:06:19:46] ENGINE Started monitor thread 'Session cleanup'.
[Wed Dec 12 06:19:46.274889 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39944] 172.18.0.5 - - [12/Dec/2018:06:19:46] "GET /idp/saml2/metadata HTTP/1.1" 200 3859 "" "curl/7.61.1"
[Wed Dec 12 06:19:46.412017 2018] [wsgi:error] [pid 1033:tid 139789161838336] [remote 172.18.0.5:39948] 172.18.0.5 - - [12/Dec/2018:06:19:46] "GET /idp/saml2/metadata HTTP/1.1" 200 3859 "" "python-requests/2.20.0"
[Wed Dec 12 06:19:46.745955 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950] [12/Dec/2018:06:19:46]  LOGIN SUCCESSFUL: admin
[Wed Dec 12 06:19:46.791054 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950] ERROR:dbus.proxies:Introspect error on :1.17:/org/freedesktop/sssd/infopipe/Users: dbus.exceptions.DBusException: org.freedesktop.DBus.Error.AuthFailed: Permission denied
[Wed Dec 12 06:19:46.791595 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950] [12/Dec/2018:06:19:46] HTTP 
[Wed Dec 12 06:19:46.791610 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950] Request Headers:
[Wed Dec 12 06:19:46.791613 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   Content-Length: 42
[Wed Dec 12 06:19:46.791616 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   HOST: idp.example.test
[Wed Dec 12 06:19:46.791619 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   ACCEPT: */*
[Wed Dec 12 06:19:46.791621 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   CONNECTION: keep-alive
[Wed Dec 12 06:19:46.791624 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   Remote-Addr: 172.18.0.5
[Wed Dec 12 06:19:46.791627 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   USER-AGENT: python-requests/2.20.0
[Wed Dec 12 06:19:46.791630 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   Content-Type: application/x-www-form-urlencoded
[Wed Dec 12 06:19:46.791632 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   ACCEPT-ENCODING: gzip, deflate
[Wed Dec 12 06:19:46.791664 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950] INFO:cherrypy.error.139789109237968:[12/Dec/2018:06:19:46] HTTP 
[Wed Dec 12 06:19:46.791669 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950] Request Headers:
[Wed Dec 12 06:19:46.791678 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   Content-Length: 42
[Wed Dec 12 06:19:46.791683 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   HOST: idp.example.test
[Wed Dec 12 06:19:46.791687 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   ACCEPT: */*
[Wed Dec 12 06:19:46.791691 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   CONNECTION: keep-alive
[Wed Dec 12 06:19:46.791695 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   Remote-Addr: 172.18.0.5
[Wed Dec 12 06:19:46.791700 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   USER-AGENT: python-requests/2.20.0
[Wed Dec 12 06:19:46.791704 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   Content-Type: application/x-www-form-urlencoded
[Wed Dec 12 06:19:46.791711 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   ACCEPT-ENCODING: gzip, deflate
[Wed Dec 12 06:19:46.793311 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950] [12/Dec/2018:06:19:46] HTTP Traceback (most recent call last):
[Wed Dec 12 06:19:46.793326 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 670, in respond
[Wed Dec 12 06:19:46.793330 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]     response.body = self.handler()
[Wed Dec 12 06:19:46.793361 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 217, in __call__
[Wed Dec 12 06:19:46.793366 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]     self.body = self.oldhandler(*args, **kwargs)
[Wed Dec 12 06:19:46.793369 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 61, in __call__
[Wed Dec 12 06:19:46.793372 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]     return self.callable(*self.args, **self.kwargs)
[Wed Dec 12 06:19:46.793375 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   File "/usr/lib/python2.7/site-packages/ipsilon/util/page.py", line 91, in __call__
[Wed Dec 12 06:19:46.793377 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]     return op(*args, **kwargs).encode('utf-8')
[Wed Dec 12 06:19:46.793380 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   File "/usr/lib/python2.7/site-packages/ipsilon/login/common.py", line 270, in root
[Wed Dec 12 06:19:46.793383 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]     return op(*args, **kwargs)
[Wed Dec 12 06:19:46.793386 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   File "/usr/lib/python2.7/site-packages/ipsilon/login/authform.py", line 29, in POST
[Wed Dec 12 06:19:46.793389 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]     return self.lm.auth_successful(self.trans, user.name, 'password')
[Wed Dec 12 06:19:46.793391 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   File "/usr/lib/python2.7/site-packages/ipsilon/login/common.py", line 155, in auth_successful
[Wed Dec 12 06:19:46.793394 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]     self.initialize_login_session(username, self.info, auth_type, userdata)
[Wed Dec 12 06:19:46.793397 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   File "/usr/lib/python2.7/site-packages/ipsilon/login/common.py", line 102, in initialize_login_session
[Wed Dec 12 06:19:46.793400 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]     infoattrs = info.get_user_attrs(username)
[Wed Dec 12 06:19:46.793403 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   File "/usr/lib/python2.7/site-packages/ipsilon/info/common.py", line 86, in get_user_attrs
[Wed Dec 12 06:19:46.793406 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]     result = p.get_user_attrs(user)
[Wed Dec 12 06:19:46.793408 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   File "/usr/lib/python2.7/site-packages/ipsilon/info/infosssd.py", line 108, in get_user_attrs
[Wed Dec 12 06:19:46.793411 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]     attrs, groups = self._get_user_data(user)
[Wed Dec 12 06:19:46.793414 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   File "/usr/lib/python2.7/site-packages/ipsilon/info/infosssd.py", line 75, in _get_user_data
[Wed Dec 12 06:19:46.793417 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]     user_path = users_if.FindByName(user)
[Wed Dec 12 06:19:46.793420 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   File "/usr/lib64/python2.7/site-packages/dbus/proxies.py", line 70, in __call__
[Wed Dec 12 06:19:46.793423 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]     return self._proxy_method(*args, **keywords)
[Wed Dec 12 06:19:46.793425 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   File "/usr/lib64/python2.7/site-packages/dbus/proxies.py", line 145, in __call__
[Wed Dec 12 06:19:46.793431 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]     **keywords)
[Wed Dec 12 06:19:46.793434 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]   File "/usr/lib64/python2.7/site-packages/dbus/connection.py", line 651, in call_blocking
[Wed Dec 12 06:19:46.793437 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950]     message, timeout)
[Wed Dec 12 06:19:46.793439 2018] [wsgi:error] [pid 1033:tid 139789228979968] [remote 172.18.0.5:39950] DBusException: org.freedesktop.DBus.Error.AuthFailed: Permission denied
```

**Comment by adelton** [comment]
Example failing run https://travis-ci.org/adelton/webauthinfra/builds/466857916.

**Comment by adelton** [comment]
Adding `,apache` to `allowed_uids` in `/etc/sssd/sssd.conf` as suggested by Jakub H. changes the error to

```
[Wed Dec 12 08:38:40.538214 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868] [12/Dec/2018:08:38:40]  LOGIN SUCCESSFUL: admin
[Wed Dec 12 08:38:40.562802 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868] [12/Dec/2018:08:38:40] HTTP 
[Wed Dec 12 08:38:40.562849 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868] Request Headers:
[Wed Dec 12 08:38:40.562861 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   Content-Length: 42
[Wed Dec 12 08:38:40.562871 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   HOST: idp.example.test
[Wed Dec 12 08:38:40.562880 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   ACCEPT: */*
[Wed Dec 12 08:38:40.562889 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   CONNECTION: keep-alive
[Wed Dec 12 08:38:40.562899 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   Remote-Addr: 172.18.0.5
[Wed Dec 12 08:38:40.562908 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   USER-AGENT: python-requests/2.20.0
[Wed Dec 12 08:38:40.562918 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   Content-Type: application/x-www-form-urlencoded
[Wed Dec 12 08:38:40.562927 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   ACCEPT-ENCODING: gzip, deflate
[Wed Dec 12 08:38:40.569331 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868] [12/Dec/2018:08:38:40] HTTP Traceback (most recent call last):
[Wed Dec 12 08:38:40.569383 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 670, in respond
[Wed Dec 12 08:38:40.569397 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]     response.body = self.handler()
[Wed Dec 12 08:38:40.569407 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 217, in __call__
[Wed Dec 12 08:38:40.569417 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]     self.body = self.oldhandler(*args, **kwargs)
[Wed Dec 12 08:38:40.569427 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 61, in __call__
[Wed Dec 12 08:38:40.569437 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]     return self.callable(*self.args, **self.kwargs)
[Wed Dec 12 08:38:40.569447 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   File "/usr/lib/python2.7/site-packages/ipsilon/util/page.py", line 91, in __call__
[Wed Dec 12 08:38:40.569457 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]     return op(*args, **kwargs).encode('utf-8')
[Wed Dec 12 08:38:40.569467 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   File "/usr/lib/python2.7/site-packages/ipsilon/login/common.py", line 270, in root
[Wed Dec 12 08:38:40.569476 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]     return op(*args, **kwargs)
[Wed Dec 12 08:38:40.569486 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   File "/usr/lib/python2.7/site-packages/ipsilon/login/authform.py", line 29, in POST
[Wed Dec 12 08:38:40.569496 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]     return self.lm.auth_successful(self.trans, user.name, 'password')
[Wed Dec 12 08:38:40.569523 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   File "/usr/lib/python2.7/site-packages/ipsilon/login/common.py", line 155, in auth_successful
[Wed Dec 12 08:38:40.569535 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]     self.initialize_login_session(username, self.info, auth_type, userdata)
[Wed Dec 12 08:38:40.569545 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   File "/usr/lib/python2.7/site-packages/ipsilon/login/common.py", line 102, in initialize_login_session
[Wed Dec 12 08:38:40.569555 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]     infoattrs = info.get_user_attrs(username)
[Wed Dec 12 08:38:40.569565 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   File "/usr/lib/python2.7/site-packages/ipsilon/info/common.py", line 86, in get_user_attrs
[Wed Dec 12 08:38:40.569575 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]     result = p.get_user_attrs(user)
[Wed Dec 12 08:38:40.569584 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   File "/usr/lib/python2.7/site-packages/ipsilon/info/infosssd.py", line 108, in get_user_attrs
[Wed Dec 12 08:38:40.569594 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]     attrs, groups = self._get_user_data(user)
[Wed Dec 12 08:38:40.569604 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   File "/usr/lib/python2.7/site-packages/ipsilon/info/infosssd.py", line 86, in _get_user_data
[Wed Dec 12 08:38:40.569614 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]     dbus_interface=dbus.PROPERTIES_IFACE)
[Wed Dec 12 08:38:40.569623 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   File "/usr/lib64/python2.7/site-packages/dbus/proxies.py", line 145, in __call__
[Wed Dec 12 08:38:40.569633 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]     **keywords)
[Wed Dec 12 08:38:40.569643 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]   File "/usr/lib64/python2.7/site-packages/dbus/connection.py", line 651, in call_blocking
[Wed Dec 12 08:38:40.569653 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868]     message, timeout)
[Wed Dec 12 08:38:40.569662 2018] [wsgi:error] [pid 956:tid 140498670622464] [remote 172.18.0.5:47868] DBusException: org.freedesktop.DBus.Error.UnknownProperty: Unknown property
```

**Comment by adelton** [comment]
With `debug_devel = 10`, `/var/log/sssd/sssd_ifp.log` contains
```
(Wed Dec 12 11:49:42 2018) [sssd[ifp]] [sbus_issue_request_done] (0x0400): org.freedesktop.DBus.Properties.Get: Success
(Wed Dec 12 11:49:42 2018) [sssd[ifp]] [sbus_dispatch] (0x4000): Dispatching.
(Wed Dec 12 11:49:42 2018) [sssd[ifp]] [sbus_method_handler] (0x2000): Received D-Bus method org.freedesktop.DBus.Properties.Get on /org/freedesktop/sssd/infopipe/Users/example_2etest/1978800000
(Wed Dec 12 11:49:42 2018) [sssd[ifp]] [sbus_senders_lookup] (0x2000): Looking for identity of sender [:1.19]
(Wed Dec 12 11:49:42 2018) [sssd[ifp]] [sbus_properties_get_send] (0x4000): Requesting property: org.freedesktop.sssd.infopipe.Users.User.extraAttributes of /org/freedesktop/sssd/infopipe/Users/example_2etest/1978800000
(Wed Dec 12 11:49:42 2018) [sssd[ifp]] [sbus_properties_get_send] (0x0040): Cannot request property org.freedesktop.sssd.infopipe.Users.User.extraAttributes [1432158310]: Unknown property
(Wed Dec 12 11:49:42 2018) [sssd[ifp]] [sbus_issue_request_done] (0x0040): org.freedesktop.DBus.Properties.Get: Error [1432158310]: Unknown property
```

**Comment by adelton** [comment]
It seems to be https://bugzilla.redhat.com/show_bug.cgi?id=1660067.

**Comment by adelton** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1660067 was fixed in Fedora 29 via sssd-2.1.0-2.fc29 and our tests in Travis CI pass with Fedora 29.
